core/src/main/groovy/noe/workspace/WorkspaceApacheDS.groovy

```diff
@@ -1,13 +1,33 @@
 package noe.workspace
 
+import groovy.util.logging.Slf4j
+import noe.common.utils.JBFile
 import noe.ews.utils.ApacheDSUtils
 import noe.server.ApacheDS
 
+@Slf4j
 class WorkspaceApacheDS extends WorkspaceAbstract {
 
+  private static final String SERVER_ID = "ews-ldap-kerberos"
+  private static final String VERSION = "2M7"
+
   WorkspaceApacheDS() {
+    prepare()
+  }
+
+  @Override
+  Object prepare() {
     basedir = ApacheDSUtils.installApacheDS(new File(basedir.toString()))
-    serverController.addServer('ews-ldap-kerberos', ApacheDS.getInstance(basedir, '2M7'))
+    serverController.addServer(SERVER_ID, ApacheDS.getInstance(basedir, VERSION))
   }
 
-}
+  @Override
+  Object destroy() {
+    serverController.removeServer(SERVER_ID)
+    File apacheDsBaseDir = new File(basedir)
+    if (apacheDsBaseDir.exists()) {
```

Comment:
The `apacheDsBaseDir` should exist for the whole life of `WorkspaceApacheDS()` so if not exist there might be at least some message about it. It is not expected state.

Reply:
Leaving messages is useless, since no one reads logs. Throwing an exception is too radical of a solution; imo we should be able to call the destroy/prepare methods idempotently. 

From the perspective of this method, I don't care what is or is not an expected state. What I care about is whether the directory exists, or not.